tachyon/zk/plonk/halo2/argument.h

```diff
@@ -125,6 +126,155 @@ class Argument {
+        BatchEvaluatePermutations(prover,
+                                  std::move(committed).TakePermutations(), x);
+
+    std::vector<std::vector<LookupEvaluated<Poly>>> evaluated_lookups_vec =
+        BatchEvaluateLookups(prover, std::move(committed).TakeLookupsVec(), x);
+
+    return {std::move(evaluated_permutations), std::move(evaluated_lookups_vec),
+            std::move(evaluated_vanishing)};
+  }
+
+  template <typename P, typename L, typename V>
+  std::vector<crypto::PolynomialOpening<Poly>> ConstructOpenings(
+      ProverBase<PCS>* prover, const ProvingKey<PCS>& proving_key,
+      const StepReturns<P, L, V>& evaluated, const F& x,
+      PointSet<F>& opening_points_set_) {
+    std::vector<RefTable<Poly>> tables = ExportPolyTables();
+    const ConstraintSystem<F>& cs =
+        proving_key.verifying_key().constraint_system();
+    opening_points_set_.Insert(x);
+
+    std::vector<crypto::PolynomialOpening<Poly>> ret;
+    ret.reserve(GetNumOpenings(proving_key, evaluated, num_circuits_));
+    for (size_t i = 0; i < num_circuits_; ++i) {
+      // Generate openings for instance of the specific circuit.
+      if constexpr (PCS::kQueryInstance) {
+        std::vector<crypto::PolynomialOpening<Poly>> openings =
+            GenerateColumnOpenings(prover, tables[i].instance_columns(),
+                                   cs.instance_queries(), x,
+                                   opening_points_set_);
+        ret.insert(ret.end(), std::make_move_iterator(openings.begin()),
+                   std::make_move_iterator(openings.end()));
+      }
+
+      // Generate openings for advice of the specific circuit.
+      std::vector<crypto::PolynomialOpening<Poly>> openings =
+          GenerateColumnOpenings(prover, tables[i].advice_columns(),
+                                 cs.advice_queries(), x, opening_points_set_);
+      ret.insert(ret.end(), std::make_move_iterator(openings.begin()),
+                 std::make_move_iterator(openings.end()));
+
+      // Generate openings for permutation of the specific circuit.
+      openings = PermutationArgumentRunner<Poly, Evals>::OpenEvaluated(
+          prover, evaluated.permutations()[i], x, opening_points_set_);
+      ret.insert(ret.end(), std::make_move_iterator(openings.begin()),
+                 std::make_move_iterator(openings.end()));
+
+      // Generate openings for lookups of the specific circuit.
+      openings = base::FlatMap(
+          evaluated.lookups_vec()[i],
+          [prover, &x, &opening_points_set_](
+              const LookupEvaluated<Poly>& evaluated_lookup) {
+            return LookupArgumentRunner<Poly, Evals>::OpenEvaluated(
+                prover, evaluated_lookup, x, opening_points_set_);
+          });
+      ret.insert(ret.end(), std::make_move_iterator(openings.begin()),
+                 std::make_move_iterator(openings.end()));
+    }
+
+    // Generate openings for fixed column.
```

Comment:
```suggestion
    // Generate openings for fixeds
```
Or for comments for advices and lookups, please add columns for them as well for consistency.